pkg/client/client.go

```diff
@@ -111,6 +129,26 @@ func (c *tlsBootstrapClientImpl) setupClientConnection(ctx context.Context) (*gr
 	return conn, nil
 }
 
+var newGetInstanceData = func(ctx context.Context, c *tlsBootstrapClientImpl, imdsURL string) (*datamodel.VMSSInstanceData, error) {
+	return c.imdsClient.GetInstanceData(ctx, baseImdsURL)
+}
+
+var newPbClientGetNonce = func(ctx context.Context, pbClient pb.AKSBootstrapTokenRequestClient, nonceRequest *pb.NonceRequest) (*pb.NonceResponse, error) {
+	return pbClient.GetNonce(ctx, nonceRequest)
+}
+
+var newPbClientGetToken = func(ctx context.Context, pbClient pb.AKSBootstrapTokenRequestClient, tokenRequest *pb.TokenRequest) (*pb.TokenResponse, error) {
+	return pbClient.GetToken(ctx, tokenRequest)
+}
+
+var newGetAttestedData = func(ctx context.Context, c *tlsBootstrapClientImpl, nonce string) (*datamodel.VMSSAttestedData, error) {
+	return c.imdsClient.GetAttestedData(ctx, baseImdsURL, nonce)
+}
+
+var newGetExecCredentialWithToken = func(token string, expirationTimestamp string) (*datamodel.ExecCredential, error) {
+	return getExecCredentialWithToken(token, expirationTimestamp)
```

Comment:
this func just takes in a couple strings and returns a struct, we can easily just throw in some placeholder values instead of having to mock it this way I think